src/components/Weather.js

```diff
@@ -0,0 +1,60 @@
+import React, { useEffect, useState } from 'react';
+import { BrowserRouter, Route, Routes } from 'react-router-dom';
+import {
+  AppShell, Navbar, Header, Burger, MediaQuery,
+} from '@mantine/core';
+import { useDispatch } from 'react-redux';
+import Cities from './Cities';
+import Details from './Details';
+import LeftNavbar from './LeftNavbar';
+import MenuBar from './MenuBar';
+import { getWeatherDetails } from '../redux/weather/DetailWeather';
+
+const Weather = () => {
+  const [opened, setOpened] = useState(false);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(getWeatherDetails('san francisco'));
+  }, []);
+
+  return (
+    <BrowserRouter>
+      <AppShell
+        padding="md"
+        navbarOffsetBreakpoint="sm"
+        asideOffsetBreakpoint="sm"
+        navbar={<Navbar height="100vh" p="md" hiddenBreakpoint="sm" hidden={!opened} width={{ sm: 200, lg: 300 }}><LeftNavbar /></Navbar>}
+        header={
+          (
+            <Header height={{ base: 50, md: 70 }} p="md">
+              <div style={{ display: 'flex', alignItems: 'center', height: '100%' }}>
+                <MediaQuery largerThan="sm" styles={{ display: 'none' }}>
+                  <Burger
+                    opened={opened}
+                    onClick={() => setOpened((o) => !o)}
+                    size="sm"
+                    mr="xl"
+                  />
+                </MediaQuery>
+
+                <MenuBar />
+              </div>
+            </Header>
+          )
+        }
+        styles={(theme) => ({
+          main: { backgroundColor: theme.colorScheme === 'dark' ? theme.colors.dark[8] : theme.colors.gray[0] },
+        })}
+      >
+        {/* Your application here */}
+        <Routes>
+          <Route path="/" element={<Cities />} />
+          <Route path="/details" element={<Details />} />
```

Comment:
- [ ] The details page is meant to be unique for each data element you're getting from the external API. To achieve this, you can use URL params with some unique detail, e.g. `id` or `name`. Something like this:

```suggestion
          <Route path="/details/:id" element={<Details />} />
```

And then, you modify the link you're using when the user clicks a city to include this parameter 🤓 🔧